From de6bca3bb04061f4d4b93a0cf22ce5daf87c3697 Mon Sep 17 00:00:00 2001
From: Artem Radchenko <artem.radchenko@globallogic.com>
Date: Thu, 26 Nov 2020 15:36:04 +0200
Subject: [PATCH 4/5] Fix NPE in AutofillManager when feature autofill is
 disabled

When feature_autofill is not present on device IAutoFillManager service could be null.
It is proposed to check service for null before trying to add client to prevent NPE.

Test: cts-tradefed run cts-dev -m CtsWidgetTestCases \
    -t android.widget.cts.DatePickerTest#testPartsVisibilityInSpinnerMode
Signed-off-by: Artem Radchenko <artem.radchenko@globallogic.com>
JiraId: ADM-5288
Change-Id: Ice3f656efad1e83ea3c9423d729c06c4b3c9b396
---
 core/java/android/view/autofill/AutofillManager.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/view/autofill/AutofillManager.java b/core/java/android/view/autofill/AutofillManager.java
index fbfeda6f0bcc..399cc897ea10 100644
--- a/core/java/android/view/autofill/AutofillManager.java
+++ b/core/java/android/view/autofill/AutofillManager.java
@@ -1912,7 +1912,7 @@ public final class AutofillManager {
             return;
         }
 
-        if (mServiceClient == null) {
+        if (mServiceClient == null && mService != null) {
             mServiceClient = new AutofillManagerClient(this);
             try {
                 final int userId = mContext.getUserId();
-- 
2.29.0

